crates/ra_editor/src/typing.rs

```diff
@@ -136,6 +136,58 @@ pub fn on_eq_typed(file: &SourceFileNode, offset: TextUnit) -> Option<LocalEdit>
     })
 }
 
+pub fn on_dot_typed(file: &SourceFileNode, offset: TextUnit) -> Option<LocalEdit> {
+    let before_dot_offset = offset - TextUnit::of_char('.');
+
+    let _whitespace = find_leaf_at_offset(file.syntax(), before_dot_offset).left_biased()?;
```

Comment:
`_` as the first symbol of var name means that the var is intentionally unused. We use this var, so we don't need `_` here and elsehwere.